packages/rpc/index.js

```diff
@@ -80,16 +100,75 @@ class SlashtagsRPC extends EventEmitter {
     return rpc
   }
 
+  /**
+   * @param {Uint8Array | string} key
+   */
+  async _rpcFromSwarm (key) {
+    /** @type {Uint8Array} */
+    const publicKey =
+      typeof key === 'string'
+        ? key.startsWith('slash')
+          ? SlashURL.parse(key).key
+          : SlashURL.decode(key)
+        : key
```

Comment:
This is really hard to read, please change. Invert will be slightly better but nested ternaries are always hard to read so I would suggest to go for `if ... else`